### ISSUE TITLE
vi-suite and Blender 3.4? Failed import of PyQT / PySide2 on MacOS

### ISSUE THREAD

**Issue by larsgrobe** [rgsouthall/vi-suite07#7]
Should the current git run on Blender 3.4, or is it currently aiming only at 3.3? On 3.4 (MacOS), when I try to activate the add-on, I get an error:
~~~
Traceback (most recent call last):
  File "/Applications/Blender.app/Contents/Resources/3.4/scripts/modules/addon_utils.py", line 333, in enable
    mod = __import__(module_name)
  File "/Users/lars/Library/Application Support/Blender/3.4/scripts/addons/vi-suite07-master/__init__.py", line 161, in <module>
    from .vi_node import vinode_categories, envinode_categories, envimatnode_categories, ViNetwork, No_Loc, So_Vi_Loc
  File "/Users/lars/Library/Application Support/Blender/3.4/scripts/addons/vi-suite07-master/vi_node.py", line 38, in <module>
    matplotlib.use('qt5agg', force=True)
  File "/Applications/Blender.app/Contents/Resources/3.4/python/lib/python3.10/site-packages/matplotlib/__init__.py", line 1166, in use
    plt.switch_backend(name)
  File "/Applications/Blender.app/Contents/Resources/3.4/python/lib/python3.10/site-packages/matplotlib/pyplot.py", line 265, in switch_backend
    backend_mod = importlib.import_module(
  File "/Applications/Blender.app/Contents/Resources/3.4/python/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/Applications/Blender.app/Contents/Resources/3.4/python/lib/python3.10/site-packages/matplotlib/backends/backend_qt5agg.py", line 7, in <module>
    from .backend_qtagg import (    # noqa: F401, E402 # pylint: disable=W0611
  File "/Applications/Blender.app/Contents/Resources/3.4/python/lib/python3.10/site-packages/matplotlib/backends/backend_qtagg.py", line 9, in <module>
    from .qt_compat import QT_API, _enum, _setDevicePixelRatio
  File "/Applications/Blender.app/Contents/Resources/3.4/python/lib/python3.10/site-packages/matplotlib/backends/qt_compat.py", line 137, in <module>
    raise ImportError(
ImportError: Failed to import any of the following Qt binding modules: PyQt6, PySide6, PyQt5, PySide2
~~~

**Comment by rgsouthall** [comment]
Hi Lars.
I have it working on 3.4 here. You still using an M1 mac?
Error implies PyQt was not installed during the activation process. Can you run Blender from the terminal - you'll get a print out of pip installing the required libraries. You may need to delete the addon folder and re-download from master to see what's going on.
Ryan

**Comment by larsgrobe** [comment]
Hi Ryan, it seams to be a problem with installing pyqt5:
~~~
Collecting PyQt5
  Using cached PyQt5-5.15.7.tar.gz (3.2 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [29 lines of output]
      Traceback (most recent call last):
        File "/Users/lars/Library/Application Support/Blender/3.4/scripts/addons/vi-suite07-master/Python/darwin/pip/_vendor/pep517/in_process/_in_process.py", line 144, in prepare_metadata_for_build_wheel
          hook = backend.prepare_metadata_for_build_wheel
      AttributeError: module 'sipbuild.api' has no attribute 'prepare_metadata_for_build_wheel'
      
      During handling of the above exception, another exception occurred:
      
      Traceback (most recent call last):
        File "/Users/lars/Library/Application Support/Blender/3.4/scripts/addons/vi-suite07-master/Python/darwin/pip/_vendor/pep517/in_process/_in_process.py", line 351, in <module>
          main()
        File "/Users/lars/Library/Application Support/Blender/3.4/scripts/addons/vi-suite07-master/Python/darwin/pip/_vendor/pep517/in_process/_in_process.py", line 333, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/Users/lars/Library/Application Support/Blender/3.4/scripts/addons/vi-suite07-master/Python/darwin/pip/_vendor/pep517/in_process/_in_process.py", line 148, in prepare_metadata_for_build_wheel
          whl_basename = backend.build_wheel(metadata_directory, config_settings)
        File "/private/var/folders/gc/k45gmlfx3gd2nl46gks5dxnr0000gn/T/pip-build-env-8q7uljq5/overlay/lib/python3.10/site-packages/sipbuild/api.py", line 46, in build_wheel
          project = AbstractProject.bootstrap('wheel',
        File "/private/var/folders/gc/k45gmlfx3gd2nl46gks5dxnr0000gn/T/pip-build-env-8q7uljq5/overlay/lib/python3.10/site-packages/sipbuild/abstract_project.py", line 87, in bootstrap
          project.setup(pyproject, tool, tool_description)
        File "/private/var/folders/gc/k45gmlfx3gd2nl46gks5dxnr0000gn/T/pip-build-env-8q7uljq5/overlay/lib/python3.10/site-packages/sipbuild/project.py", line 585, in setup
          self.apply_user_defaults(tool)
        File "/private/var/folders/gc/k45gmlfx3gd2nl46gks5dxnr0000gn/T/pip-install-36kb3x2j/pyqt5_4d7bc95fd9b34548bbc3b24f89cc42ea/project.py", line 69, in apply_user_defaults
          super().apply_user_defaults(tool)
        File "/private/var/folders/gc/k45gmlfx3gd2nl46gks5dxnr0000gn/T/pip-build-env-8q7uljq5/overlay/lib/python3.10/site-packages/pyqtbuild/project.py", line 70, in apply_user_defaults
          super().apply_user_defaults(tool)
        File "/private/var/folders/gc/k45gmlfx3gd2nl46gks5dxnr0000gn/T/pip-build-env-8q7uljq5/overlay/lib/python3.10/site-packages/sipbuild/project.py", line 236, in apply_user_defaults
          self.builder.apply_user_defaults(tool)
        File "/private/var/folders/gc/k45gmlfx3gd2nl46gks5dxnr0000gn/T/pip-build-env-8q7uljq5/overlay/lib/python3.10/site-packages/pyqtbuild/builder.py", line 69, in apply_user_defaults
          raise PyProjectOptionException('qmake',
      sipbuild.pyproject.PyProjectOptionException
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
~~~

**Comment by rgsouthall** [comment]
Looks like an M1 problem with a certain version of PyQt5. PyQt6 may solve the problem but causes others. Can you try replacing PyQt5 with PyQt5==5.15.6 on line 119 of the __init__.py file?

**Comment by larsgrobe** [comment]
Hi Ryan,

I just changed all occurances of `PyQT5` to `PyQT6` in `__init__.py`, `vi_operators.py`, and `vi_display.py`. This allows me to enable the 0.7 master branch on Blender 3.6. I got a sun path displayed, so the GUI basically seems to work. This may bring back vi-suite to current (arm-based) Macs.

With PyQT5 there is apparently little hope, since there is no official backport of the old versions to Apple's arm platform. One might try to install ports by homebrew etc. and replace Blender's python environment, but that looks hard to maintain.

Best, Lars.
